tests/e2e/gc_rwx_multi_gc.go

```diff
@@ -0,0 +1,779 @@
+		3. Wait for PVC to be in the Bound phase
+		4. Verify CnsVolumeMetadata CRD is created
+		5. Verify if the mapping PVC is bound in the SV cluster using the volume handler
+		6. Verify health status of PVC
+		7. Verify volume is created on CNS by using CNSQuery API and also check metadata is pushed to CNS
+		8. Create a Pod1 using PVC created above at a mount path specified in PodSpec
+		9. Verify CnsFileAccessConfig CRD is created
+		10. Verify Pod1 is in the Running phase
+		11. Create a file (file1.txt) at the mount path. Check if the creation is successful
+
+		From a new guest cluster (Guest cluster2 )in the same wcp namespace, do the following:
+		12. Create PV with VolumeHandle from supervisor cluster's PVC in the GC
+		13. Verify CnsVolumeMetadata CRD is created
+		14. Wait for PV entry to be present in CNS
+		15. Verify if the mapping PVC is bound in the SV cluster using the volume handler
+		16. Verify health status of PVC
+		17. Verify volume is created on CNS by using CNSQuery API and also check metadata is pushed to CNS
+		18. Create a Pod2 using PVC created above at a mount path specified in PodSpec
+		19. Verify CnsFileAccessConfig CRD is created
+		20. Verify Pod2 is in the Running phase
+		21. Verify ACL net permission set by calling CNSQuery for the file volume
+		22. Create a file (file1.txt) at the mount path. Check if the creation is successful
+		23. Delete Pod2
+		24. Verify CnsFileAccessConfig CRD is deleted
+		25. Verify if the Pod is successfully deleted
+		26. Cleanup PV & PVC from Guest cluster1 and Guest Cluster2
+		28. Delete PVC from GCs
+		29. Verify if PVC is deleted in GCs
+		30. Delete PVs in GCs and SVCs
+		31. Verify if PVs are deleted in the GCs and SVCs
+		32. Verify CnsVolumeMetadata CRD is deleted
+		33. Check if the VolumeID is deleted from CNS by using CNSQuery API
+	*/
+	ginkgo.It("[rwm-csi-tkg] Verify static provisioning across Guest Clusters within the "+
+		"same WCP namespace - part 1", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		newGcKubconfigPath := os.Getenv("NEW_GUEST_CLUSTER_KUBE_CONFIG")
+		if newGcKubconfigPath == "" {
+			ginkgo.Skip("Env NEW_GUEST_CLUSTER_KUBE_CONFIG is missing")
+		}
+
+		var storageclasspvc *storagev1.StorageClass
+		var pvclaim *v1.PersistentVolumeClaim
+		var pv *v1.PersistentVolume
+		var err error
+		defaultDatastore = getDefaultDatastore(ctx)
+
+		ginkgo.By("CNS_TEST: Running for GC setup")
+		scParameters[svStorageClassName] = storagePolicyName
+		ginkgo.By("Creating a PVC")
+		storageclasspvc, pvclaim, err = createPVCAndStorageClass(client,
+			namespace, nil, scParameters, diskSize, nil, "", false, v1.ReadWriteMany)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		pvcUID := string(pvclaim.GetUID())
+		framework.Logf("PVC UUID in GC " + pvcUID)
+
+		defer func() {
+			err = client.StorageV1().StorageClasses().Delete(ctx, storageclasspvc.Name, *metav1.NewDeleteOptions(0))
```

Comment:
run linter check locally

Reply:
rpanduranga@rpanduranga-a01 vsphere-csi-driver % make check
hack/check-format.sh
hack/check-mdlint.sh
hack/check-shell.sh
hack/check-staticcheck.sh
hack/check-vet.sh
hack/check-golangci-lint.sh
golangci/golangci-lint info checking GitHub for tag 'v1.40.1'
golangci/golangci-lint info found version: 1.40.1 for v1.40.1/darwin/amd64
golangci/golangci-lint info installed /Users/rpanduranga/go/bin/golangci-lint
INFO [config_reader] Config search paths: [./ /Users/rpanduranga/CSI-Code/vsphere-csi-driver /Users/rpanduranga/CSI-Code /Users/rpanduranga /Users /] 
INFO [config_reader] Used config file .golangci.yml 
INFO [lintersdb] Active 12 linters: [deadcode errcheck gosimple govet ineffassign lll misspell staticcheck structcheck typecheck unused varcheck] 
INFO [loader] Go packages loading at mode 575 (files|imports|name|types_sizes|compiled_files|deps|exports_file) took 1.91261738s 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 91.234424ms 
INFO [linters context/goanalysis] analyzers took 0s with no stages 
INFO [runner] Issues before processing: 110, after processing: 0 
INFO [runner] Processors filtering stat (out/in): identifier_marker: 21/21, skip_files: 110/110, nolint: 0/1, filename_unadjuster: 110/110, path_prettifier: 110/110, skip_dirs: 110/110, autogenerated_exclude: 21/110, exclude: 21/21, exclude-rules: 1/21, cgo: 110/110 
INFO [runner] processing took 11.390271ms with stages: nolint: 9.42051ms, autogenerated_exclude: 1.001892ms, path_prettifier: 482.254µs, identifier_marker: 237.268µs, skip_dirs: 126.69µs, exclude-rules: 93.214µs, cgo: 12.458µs, filename_unadjuster: 11.632µs, max_same_issues: 1.101µs, uniq_by_line: 574ns, max_from_linter: 420ns, diff: 359ns, source_code: 356ns, skip_files: 307ns, exclude: 238ns, max_per_file_from_linter: 230ns, path_shortener: 218ns, severity-rules: 216ns, sort_results: 210ns, path_prefixer: 124ns 
INFO [runner] linters took 2.461944704s with stages: goanalysis_metalinter: 2.450458395s 
INFO File cache stats: 0 entries of total size 0B 
INFO Memory: 46 samples, avg is 111.0MB, max is 140.9MB 
INFO Execution took 4.477375874s